crates/runtime/src/dataconnector/azure.rs

```diff
@@ -0,0 +1,256 @@
+        .default("false"),
+    ParameterSpec::connector("use_fabric_endpoint")
+        .description("Use the Azure Storage fabric endpoint.")
+        .default("false"),
+    ParameterSpec::connector("allow_http")
+        .description("Allow insecure HTTP connections.")
+        .default("false"),
+    ParameterSpec::connector("authority_host")
+        .description("Sets an alternative authority host."),
+    ParameterSpec::connector("max_retries")
+        .description("The maximum number of retries.")
+        .default("3"),
+    ParameterSpec::connector("retry_timeout")
+        .description("The retry timeout."),
+    ParameterSpec::connector("backoff_initial_duration")
+        .description("The initial backoff duration."),
+    ParameterSpec::connector("backoff_max_duration")
+        .description("The maximum backoff duration."),
+    ParameterSpec::connector("backoff_base")
+        .description("The base of the exponential to use"),
+    ParameterSpec::connector("proxy_url")
+        .description("The proxy URL to use."),
+    ParameterSpec::connector("proxy_ca_certificate")
+        .description("The CA certificate for the proxy."),
+    ParameterSpec::connector("proxy_excludes")
+        .description("Set list of hosts to exclude from proxy connections"),
+    ParameterSpec::connector("msi_endpoint")
+        .description("Sets the endpoint for acquiring managed identity tokens."),
+    ParameterSpec::connector("federated_token_file")
+        .description("Sets a file path for acquiring Azure federated identity token in Kubernetes"),
+    ParameterSpec::connector("use_cli")
+        .description("Set if the Azure Cli should be used for acquiring access tokens."),
+    ParameterSpec::connector("skip_signature")
+        .description("Skip fetching credentials and skip signing requests. Used for interacting with public containers."),
+    ParameterSpec::connector("disable_tagging")
+        .description("Ignore any tags provided to put_opts"),
+
+    // Common listing table parameters
+    ParameterSpec::runtime("file_format"),
+    ParameterSpec::runtime("file_extension"),
+    ParameterSpec::runtime("csv_has_header")
+        .description("Set true to indicate that the first line is a header."),
+    ParameterSpec::runtime("csv_quote").description("The quote character in a row."),
+    ParameterSpec::runtime("csv_escape").description("The escape character in a row."),
+    ParameterSpec::runtime("csv_schema_infer_max_records")
+        .description("Set a limit in terms of records to scan to infer the schema."),
+    ParameterSpec::runtime("csv_delimiter")
+        .description("The character separating values within a row."),
+    ParameterSpec::runtime("file_compression_type")
+        .description("The type of compression used on the file. Supported types are: GZIP, BZIP2, XZ, ZSTD, UNCOMPRESSED"),
+];
+
+pub const PREFIXES: [&'static str; 6] = ["azure", "abfs", "abfss", "adl", "adls", "az"];
```

Comment:
You shouldn't need to do this if the main prefix is `azstorage`. These should be proper schema prefixes E.g. `abfs://` and on false, an error should be thrown, not just a continuation.